### ISSUE TITLE
`sidre::Group::save()` generates invalid json for empty arrays

### ISSUE THREAD

**Issue by kennyweiss** [LLNL/axom#1018]
When using `json` protocols, `sidre::Group::save()` generates invalid json for views/buffers associated with empty arrays.

Specifically, empty arrays will produce entries with a key that's missing a value, e.g. 
```
value: 
```
but should probably generate a value that has an empty array, e.g. 
```
value: []
```

I'm reasonably confident that the problem is in the underlying conduit calls, e.g. `conduit::Node::to_json_generic()`.

#### Reproducer:
If we change the following line from `sidre_hdf5` to `sidre_json`, the `spio_parallel` tests fail when loading the written datastore.
https://github.com/LLNL/axom/blob/19af5ad324550ef7cb969bc8cf4f57cd7b575a94/src/axom/sidre/tests/spio/spio_main.cpp#L16

Here is an example of the generated output.
<details>
  <summary>Expand to see full json output </summary>

```json
{
  "sidre": 
  {
    "groups": 
    {
      "fields": 
      {
        "groups": 
        {
          "a": 
          {
            "views": 
            {
              "i0": 
              {
                "state": "SCALAR",
                "value": 303
              }
            }
          }
        }
      },
      "fields2": 
      {
        "groups": 
        {
          "b": 
          {
            "views": 
            {
              "i1": 
              {
                "state": "BUFFER",
                "buffer_id": 0,
                "schema": "{\"dtype\":\"int32\",\"number_of_elements\": 10,\"offset\": 0,\"stride\": 4,\"element_bytes\": 4,\"endianness\": \"little\"}",
                "is_applied": 1
              }
            }
          },
          "c": 
          {
            "views": 
            {
              "i2": 
              {
                "state": "BUFFER",
                "buffer_id": 1,
                "schema": "{\"dtype\":\"int32\",\"number_of_elements\": 0,\"offset\": 0,\"stride\": 4,\"element_bytes\": 4,\"endianness\": \"little\"}",
                "is_applied": 1
              }
            }
          }
        }
      }
    },
    "buffers": 
    {
      "buffer_id_0": 
      {
        "id": 0,
        "schema": "{\"dtype\":\"int32\",\"number_of_elements\": 10,\"offset\": 0,\"stride\": 4,\"element_bytes\": 4,\"endianness\": \"little\"}",
        "data": [4010, 4400, 4788, 5174, 5558, 5940, 6320, 6698, 7074, 7448]
      },
      "buffer_id_1": 
      {
        "id": 1,
        "schema": "{\"dtype\":\"int32\",\"number_of_elements\": 0,\"offset\": 0,\"stride\": 4,\"element_bytes\": 4,\"endianness\": \"little\"}",
        "data": 
      }
    },
    "attribute": 
    {
    },
    "external": 
    {
    }
  },
  "sidre_group_name": ""
}
```
</details>

The problem is near the bottom -- `buffer_id_1` is empty (i.e. it has 0 elements) and the corresponding `data` field is empty.
I think it should look like this:
```diff
      "buffer_id_1": 
      {
        "id": 1,
        "schema": "{\"dtype\":\"int32\",\"number_of_elements\": 0,\"offset\": 0,\"stride\": 4,\"element_bytes\": 4,\"endianness\": \"little\"}",
-        "data": 
+        "data": []
      }
```

**Comment by nselliott** [comment]
I think this is the underlying reason for a comment I made a while back in #97 .  

> Another case where save/load round trips do not produce equivalence is that some protocols (non sidre_*) do not preserve an empty sub-Group through a save/load operation.

**Comment by kennyweiss** [comment]
Great news -- this bug has already been resolved in `conduit@0.8.4:`!
I'm updating conduit to `v0.8.6` in a separate branch.
